src/state/slices/opportunitiesSlice/selectors.test.ts

```diff
@@ -0,0 +1,272 @@
+        serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractOne'),
+        serializeUserStakingId('eip155:1:0xfauxmes', 'eip155:1:0xLpContractOne'),
+      ] as UserStakingId[],
+      byId: {
+        [serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractTwo')]: {
+          stakedAmountCryptoPrecision: '1337',
+          rewardsAmountCryptoPrecision: '420',
+        },
+        [serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractOne')]: {
+          stakedAmountCryptoPrecision: '4',
+          rewardsAmountCryptoPrecision: '3',
+        },
+        [serializeUserStakingId('eip155:1:0xfauxmes', 'eip155:1:0xLpContractOne')]: {
+          stakedAmountCryptoPrecision: '9000',
+          rewardsAmountCryptoPrecision: '1',
+        },
+      },
+    }
+
+    const mockState = {
+      ...baseState,
+      portfolio: {
+        ...baseState.portfolio,
+        accountBalances,
+      },
+      opportunities: {
+        ...initialState,
+        lp,
+        staking,
+        userStaking,
+      },
+    }
+    describe('selectLpOpportunityIdsByAccountId', () => {
+      it('can get LP opportunities Ids for a given AccountId', () => {
+        const result = selectLpOpportunityIdsByAccountId(mockState, {
+          accountId: 'eip155:1:0xgomes',
+        })
+
+        expect(result).toEqual(['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'])
+      })
+    })
+    describe('selectStakingOpportunityIdsByAccountId', () => {
+      it('can get staking opportunities Ids for a given AccountId', () => {
+        const result = selectStakingOpportunityIdsByAccountId(mockState, {
+          accountId: 'eip155:1:0xgomes',
+        })
+
+        expect(result).toEqual(['eip155:1:0xLpContractOne', 'eip155:1:0xLpContractTwo'])
+      })
+    })
+    describe('selectHighestBalanceLpUserStakingIdByStakingId', () => {
+      it('can get the highest balance AccuontId for a given StakingId', () => {
+        const result = selectHighestBalanceAccountIdIdByStakingId(mockState, {
+          stakingId: 'eip155:1:0xLpContractOne',
+        })
+
+        expect(result).toEqual('eip155:1:0xfauxmes')
+      })
+    })
+    describe('selectHighestBalanceAccountIdIdByLpId', () => {
+      it('can get the highest balance AccountId for a given LpId', () => {
+        const result = selectHighestBalanceAccountIdIdByLpId(mockState, {
+          lpId: 'eip155:1:0xLpContractOne',
+        })
+
+        expect(result).toEqual('eip155:1:0xfauxmes')
+      })
+    })
+  })
+  describe('selectUserStakingOpportunityByUserStakingId', () => {
+    const staking = {
+      ...initialState.staking,
+    }
+    const userStaking = {
+      ...initialState.staking,
+      ids: [
+        serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractTwo'),
+        serializeUserStakingId('eip155:1:0xgomes', 'eip155:1:0xLpContractOne'),
+        serializeUserStakingId('eip155:1:0xfauxmes', 'eip155:1:0xLpContractOne'),
+      ] as UserStakingId[],
```

Comment:
unnecessary cast?

Reply:
Nice yeah, the util will now give us the proper return value - d2d2ba4b5b827a0097b7cd4b62f93732218d4df7